### DISCUSSION TITLE
How to display products custom fields at order details?

### DISCUSSION THREAD

**Discussion by eduarddotgg** [vendurehq/vendure#1589] [Q&A]
is there any easy/proper way to display products custom fields at order details? if there is any example that would be very helpful.

**Comment by michaelbromley** [comment]
Hi, something like this:
```gql
{
  activeOrder {
    id
    code 
    # ... etc
    lines {
      id
      # .. etc
      productVariant {
        product {
          customFields {
            #... add your custom field selection here
          }
        }
      }
    }
  }
}
```

**Reply by eduarddotgg** [reply]
Thanks! But I'm sorry if forgat to mention that I would like to show product's custom fields at admin ui's order page. I'm sorry, my bad.

**Reply by michaelbromley** [reply]
Ah, ok. No built-in way to do that, so you'd need to implement a custom component to replace the order list/detail views using ui extensions. Some general docs here: https://www.vendure.io/docs/plugins/extending-the-admin-ui/

In particular, you'd need to define your own list & detail routes and then [replace the default route in the nav menu](https://www.vendure.io/docs/plugins/extending-the-admin-ui/adding-navigation-items/#overriding-existing-items).

**Reply by eduarddotgg** [reply]
ok, I see. thanks!
